runtime/caching/cache.go

```diff
@@ -111,9 +120,14 @@ func (c *Cache) Require(ctx Context, options interface{}) (*Handle, error) {
 		c.entries = append(c.entries, entry)
 
 		go entry.created.Do(func() {
-			defer entry.ctx.dispose() // ensure resources are cleanup when constructor is done
+			defer entry.ctx.dispose()
+			// ensure resources are cleanup when constructor is done
+
+			c.constructortime.Time("constructor-creation-time", func() {
```

Comment:
Use same `monitor` as given in `New`, then call `monitor.Time('constructor-duration', func() {...`